server/src/main/java/org/opensearch/action/update/UpdateHelper.java

```diff
@@ -128,7 +129,11 @@ Tuple<UpdateOpType, Map<String, Object>> executeScriptedUpsert(Map<String, Objec
 
         if (operation != UpdateOpType.CREATE && operation != UpdateOpType.NONE) {
             // Only valid options for an upsert script are "create" (the default) or "none", meaning abort upsert
-            logger.warn("Invalid upsert operation [{}] for script [{}], doing nothing...", operation, script.getIdOrCode());
+            if (logger.isDebugEnabled() || ScriptType.STORED.equals(script.getType())) {
```

Comment:
why check for stored script here?

Reply:
In case of stored script, id gets logged which should be fine as it shouldn't expose user data and continues to help with debugging. For inline scripts in request, we may end up logging the entire user provided script here. Hence, kept this check to ensure some debuggability,

Reply:
make sense, thanks